web/src/api/getMeta.ts

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from '@tanstack/react-query';
+import { getBasePath, QUERY_KEYS } from 'api/helpers';
+
+export type callerLocation = [number, number] | null;
+
+export interface Meta {
+  features?: Record<string, boolean>;
+  callerLocation?: callerLocation;
+}
+
+export async function getMeta(): Promise<Meta> {
+  const path: URL = new URL(`/v8/${QUERY_KEYS.META}`, getBasePath());
+
+  try {
+    const response = await fetch(path);
+    if (response.ok) {
+      const data = await response.json();
+      return data;
+    }
+
+    throw new Error(await response.text());
+  } catch (error) {
+    // If the request fails, we will return an empty object instead of throwing an error
+    // as the is still functional without the data
+    console.error(error);
+    return {};
+  }
+}
+
+export function useMeta(): Meta {
+  return (
+    useQuery<Meta>([QUERY_KEYS.META], async () => getMeta(), {
+      suspense: true,
+    }).data ?? {}
+  );
+}
```

Comment:
Not sure if it should be here or further downstream but should we guard against localhost/non valid IPs returning Nan as indicated by this sentry issue?

https://electricitymaps.sentry.io/issues/5092639410/?project=4504366922989568&query=is%3Aunresolved+nan&referrer=issue-stream&statsPeriod=90d&stream_index=0

Would be nice to eliminate those when we are tinkering with it.

Reply:
Good idea, done :)

Reply:
Thanks! Now let's just hope that is the actual cause 😅

Reply:
Seems like it was mostly coming from a "Facebook Bot" browser, so I assume it's not really affecting any users :)